### PR TITLE
[RFC 0010] Nixpkgs development support

### PR DESCRIPTION
**Pull request by shlevy**
Rendered:  https://github.com/shlevy/rfcs/blob/nixpkgs-development-support/rfcs/0010-nixpkgs-development-support.md

### CODE REVIEW

**Comment by shlevy** [comment]
@edolstra @peti @domenkozar @globin

**Comment by edolstra** [comment]
This sounds like it could easily become a dumping ground of random crap. What would be the inclusion criteria? The alternative - using separate repositories - seems fine for development since people can use `builtins.fetchgit` to get them. If some tool becomes popular, it can always be merged into the Nixpkgs repo.

Also, I'm afraid `builtins.exec` may become a Pandora's box turning Nix into an imperative language.

**Comment by shlevy** [comment]
I'm reluctant to assume the availability of `builtins.fetchgit` until 1.12 is released.

We don't have explicit inclusion criteria elsewhere in nixpkgs, do we? It should be something generally usefuil and maintainable and up to basic standards of code cleanliness.

Alternate repositories can't be kept in tight lockstep with nixpkgs. This is why we have nixos in nixpkgs, right?

`builtins.exec` is only turned on if you ask for it, and it solves some significant real problems. Do you have an alternative?

**Comment by edolstra** [comment]
Another issue: How does this differ from `pkgs/build-support`?

File: rfcs/0010-nixpkgs-development-support.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+# Detailed design
+[design]: #detailed-design
+
+Add `pkgs/development-support`, add `pkgs.development-support` calling
+the functions defined in `pkgs/development-support`.
+
+Any tool relying on `builtins.exec` will take that function as an
+argument, rather than using the builtin directly, so that all of
+nixpkgs evaluates properly without
+`allow-unsafe-native-code-during-evaluation`.
```
Comment (edolstra) [inline rfcs/0010-nixpkgs-development-support.md]: I don't see why this is necessary.

File: rfcs/0010-nixpkgs-development-support.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+the `builtins.exec` nix feature, there are a number of developer tools
+I'd like to make widely available, such as something to translate a
+.gitignore into a filterSource call or do no-sha256 fetchgit with
+caching. It would be good to have a single blessed place to put these
+kinds of tools.
+
+# Detailed design
+[design]: #detailed-design
+
+Add `pkgs/development-support`, add `pkgs.development-support` calling
```
Comment (edolstra) [inline rfcs/0010-nixpkgs-development-support.md]: -> `pkgs.developmentSupport`

**Comment by shlevy** [comment]
@edolstra `build-support` is useful things that help nix builds. development-support is mostly evaluation time stuff.

File: rfcs/0010-nixpkgs-development-support.md (hunk 1 above)
Comment (shlevy) [inline rfcs/0010-nixpkgs-development-support.md]: It makes it possible with a grep (possibly even enforced before merging) that nothing you build in nixpkgs can cause a problem if you have `allow-unsafe-native-code-during-evaluation` on.

File: rfcs/0010-nixpkgs-development-support.md (hunk 2 above)
Comment (shlevy) [inline rfcs/0010-nixpkgs-development-support.md]: Right, fixing

**Comment by copumpkin** [comment]
I think it's partly about discoverability and partly about co-versioned-ness. In many ways the closer analog to what you're talking about feels like the maintainers folder and possibly @aszlig's md5 collision abomination. These both strongly suggest to me that we want tooling for working on the nixpkgs repo, and want many of the goodies that Nix provides when working on Nixpkgs itself

**Comment by shlevy** [comment]
@copumpkin Want to coauthor?

**Comment by shlevy** [comment]
Created https://github.com/shlevy/nixpkgs-development-support for now

File: rfcs/0010-nixpkgs-development-support.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+# Motivation
+[motivation]: #motivation
+
+Currently, nixpkgs has some features, like `callCabal2nix` and the
+`env` attribute of haskell packages, that don't really have anything
+to do with package management per se but rather provide support for
+developers who use nix to manage their projects. With the merging of
+the `builtins.exec` nix feature, there are a number of developer tools
+I'd like to make widely available, such as something to translate a
+.gitignore into a filterSource call or do no-sha256 fetchgit with
```
Comment (michalrus) [inline rfcs/0010-nixpkgs-development-support.md]: :heart_eyes:

**Comment by 0xABAB** [comment]
Another feature I would not want to see. Others have given enough arguments already.

**Pull request closed without merging**